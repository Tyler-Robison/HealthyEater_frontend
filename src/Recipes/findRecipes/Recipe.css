.Recipe {
    background-color: white;
    border: 2px solid black;
    min-width: 450px;
    margin: 1rem 0;
    border-radius: 2%;
    box-shadow: 10px 5px 5px rgb(87, 87, 87);
}

.Recipe-img {
    margin-bottom: 1rem;
}

@media (max-width: 468px) {

    .Recipe {
        font-size: 12px;
        min-width: 300px;
    }

    .Recipe-img {
        max-width: 200px;
    }
}

@media (max-width: 468px) {

    .Recipe {
        font-size: 12px;
        min-width: 300px;
    }

    .Recipe-img {
        max-width: 200px;
    }
}

