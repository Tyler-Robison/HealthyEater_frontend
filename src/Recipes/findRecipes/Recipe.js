import React from "react";
import { Link } from 'react-router-dom';

const Recipe = ({ recipe }) => {

    const displayIngredients = recipe => recipe.usedIngredients.map((ingredient, idx) => {
        // makes sure no comma after last item
        return (idx === recipe.usedIngredients.length - 1 ? `${ingredient.name}` : `${ingredient.name}, `);
    })

    return (
        <div key={recipe.id} className='RecipeList-div'>
            <Link to={`/${recipe.id}/detail`} className='RecipeList-para'><b>{recipe.title}</b></Link>
            <p>Uses {recipe.usedIngredients.length} of your ingredients ( {displayIngredients(recipe)} )</p>
            <img className='RecipeList-img' src={recipe.image} alt={`Picture of the recipe ${recipe.title}`}></img>
        </div>
    )
}

export default Recipe;