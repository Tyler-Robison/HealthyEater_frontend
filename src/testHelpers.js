const mealplanArr = [
    { id: 1, day: 'Mon', recipe_id: 1111, name: 'bacon', ww_points: 20 },
    { id: 2, day: 'Mon', recipe_id: 2222, name: 'eggs', ww_points: 10 },
    { id: 3, day: 'Tues', recipe_id: 3333, name: 'salad', ww_points: 5 }
];

const recipeArr = [
    { name: 'bacon', recipe_id: 1111, ww_points: 20 },
    { name: 'eggs', recipe_id: 2222, ww_points: 10 },
    { name: 'salad', recipe_id: 3333, ww_points: 5 },
    { name: 'bread', recipe_id: 4444, ww_points: 7 },
];

// mock currentUser
const testUser = {
    id: 1,
    username: "Tyler",
    isAdmin: false,
    points: null,
    mealplan: mealplanArr,
    recipes: recipeArr
};

// mock recipe search results
const testRecipes = [
    {
        id: 1234,
        title: 'Scrambled Eggs',
        usedIngredients: [{ name: 'eggs' }, { name: 'otherIngredient' }],
        image: 'https://spoonacular.com/recipeImages/716426-312x231.jpg'
    },
    {
        id: 5678,
        title: 'BLT',
        usedIngredients: [{ name: 'bacon' }, { name: 'lettuce' }, { name: 'tomato' }],
        image: 'https://spoonacular.com/recipeImages/716426-312x231.jpg'
    }
];

export {
    testUser,
    testRecipes
}