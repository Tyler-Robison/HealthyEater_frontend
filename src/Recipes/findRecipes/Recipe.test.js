import React from 'react';
import { render, screen } from '@testing-library/react';
import Recipe from './Recipe';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext'
import { testRecipes } from '../../testHelpers';

// using listState to set 
const DummyComponent = () => {

    return (
        <Recipe recipe={testRecipes[0]} />
    )
}

it("renders without crashing", function () {
    <MemoryRouter>
        <ContextProvider >
            <DummyComponent />
        </ContextProvider>
    </MemoryRouter>
});

it("matches snapshot", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <DummyComponent />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});

it("correctly displays recipe values", function () {
    render(
        <MemoryRouter>
            <ContextProvider >
                <DummyComponent />
            </ContextProvider>
        </MemoryRouter>
    );

    // name
    expect(screen.getByText('Scrambled Eggs')).toBeInTheDocument()
    
    // ingredient count
    expect(screen.getByText('Uses 2 of your ingredients ( eggs, otherIngredient )')).toBeInTheDocument()
})