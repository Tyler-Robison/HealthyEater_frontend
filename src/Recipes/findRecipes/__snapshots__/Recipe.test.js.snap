// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="RecipeList-div"
  >
    <a
      class="RecipeList-para"
      href="/1234/detail"
    >
      <b>
        Scrambled Eggs
      </b>
    </a>
    <p>
      Uses 2 of your ingredients ( eggs, otherIngredient )
    </p>
    <img
      alt="Picture of the recipe Scrambled Eggs"
      class="RecipeList-img"
      src="https://spoonacular.com/recipeImages/716426-312x231.jpg"
    />
  </div>
</DocumentFragment>
`;
